src/Introspector.php

```diff
@@ -21,12 +22,11 @@ public function introspect(): void
     {
         $client = $this->endpointConfig->makeClient();
 
-        $introspectionResult = $client->request(
-            Introspection::getIntrospectionQuery([
-                'directiveIsRepeatable' => true,
-            ])
-        );
-        $introspectionResult->assertErrorFree();
+        try {
+            $introspectionResult = $this->fetchIntrospectionResult($client, true);
+        } catch (Throwable $_) {
```

Comment:
Maybe we could catch the `InvalidResponseException` right away?

Reply:
That exception comes from the client, which is configurable and may throw any other exception as well.